refactor(init, manager): move package installation from `init` to `manager`

angeloashmore opened the pull request:
<!--- Provide a general summary of your changes in the title above -->

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [x] Chore (a non-breaking change which is related to package maintenance)
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

> **Note**: This PR is a work in progress and may be rejected.

This PR refactors `@slicemachine/init` and `@slicemachine/manager` to move package installation logic from `init` to `manager`.

The refactor was done for a few reasons:

1. Package manager detection is project-level metadata that could be used elsewhere in Slice Machine. As such, it makes sense to make it a method in `ProjectManager` (`manager.project.detectPackageManager()`).
2. We may want to handle package installation in other locations, such as lazily in plugins (think: only install a dependency once it is needed).
3. Project initialization via `@slicemachine/init` requires a few steps, including installing core dependencies (SM, an adapter). It also involves installing a plugin's dependencies and letting a plugin bootstrap any files it needs.
   - We currently couple this step with `@slicemachine/init` using the `command:init` plugin hook.
   - However, we could make it more general by naming it `project:init` and make it something `@slicemachine/manager` handles. That allows us to run the initialization again outside the process of `@slicemachine/init`.

This refactor is mostly complete except that `@slicemachine/init`'s tests don't pass (so maybe it's not mostly done). `execa` doesn't seem to be mocked in `@slicemachine/manager`, which causes package installation to actually happen on the filesystem. If you run the tests, be aware that a `foo` package will be installed into `@slicemachine/init`.

## Checklist:

<!--- Put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] My change requires an update to the official documentation.
- [ ] All [TSDoc](https://tsdoc.org) comments are up-to-date and new ones have been added where necessary.
- [ ] All new and existing tests are passing.

<!--- A cute animal (or car) picture is welcome to close your PR! -->

🦬

angeloashmore committed:
refactor(init, manager): move package installation from `init` to `manager` (WIP)

lihbr reviewed (approved):
LGTM, let's get those tests to work

lihbr committed:
refactor(plugin-kit): `command:init` is now `project:init`

lihbr committed:
test(init): fix test following install refactor

lihbr committed:
chore(deps): update vitest

lihbr committed:
style: please linter

lihbr committed:
chore(deps): fix dependencies

lihbr committed:
fix(manager): typing

lihbr committed:
chore(deps): pin vitest

This prevents https://github.com/vitest-dev/vitest/issues/2806 which is broken in the latest version of vitest for now

lihbr committed:
test(init): types

lihbr committed:
test(deps): wrong dependency declaration

lihbr committed:
Merge branch 'dev-plugins' of github.com:prismicio/slice-machine into aa/dev-plugins-refactor-dep-install

lihbr committed:
test(manager): move project init tests to manager

lihbr committed:
refactor(adapter-next): `command:init` > `project:init`

lihbr committed:
build(manager): inline execa 7

lihbr committed:
style: lint
